libs/sdk-ui-pivot/src/impl/agGridDataSource.ts

```diff
@@ -216,3 +231,33 @@ function isSortedByFirstAttribute(headers: TableHeaders, sortModel: ISortModelIt
 
     return headers.allHeaders[0].field === sortModel[0].colId;
 }
+
+function isDataViewSortedByFirstAttribute(dv: DataViewFacade): boolean {
+    const { sortBy } = dv.definition;
+    if (!sortBy || !sortBy.length) {
+        // this is somewhat dangerous assumption: no explicit sort == sorted by first col
+        //  (bear backend behaves thusly)
+        return true;
+    }
+
+    if (sortBy.length > 1) {
+        return false;
+    }
+
+    const firstSortBy = sortBy[0];
+    if (!isAttributeSort(firstSortBy)) {
+        return false;
+    }
+
+    const attributeBucket = dv.def().bucket(BucketNames.ATTRIBUTE);
+    if (!attributeBucket) {
+        return false;
+    }
+
+    const firstAttribute = bucketAttributes(attributeBucket)[0];
+    if (!firstAttribute) {
+        return false;
+    }
+
+    return firstSortBy.attributeSortItem.attributeIdentifier === attributeLocalId(firstAttribute);
+}
```

Comment:
I think this method is better located in DataViewFacade itself; in the IExecutionDefinitionMethods. then do dv.def().isSortedByFirstAttribute().

the safest approach would be to move this to dv.meta() and combine with the effectiveSortItems() (however this method concerns itself with measure sorts only.. it will need enhancement) - but I suggest to leave this for now.

Reply:
I think this is specific to table because it uses the ATTRIBUTE bucket explicitly. We could make it in the facade and provide some parameter to say attributes from which buckets to use, but I still think this query (sorted by first attribute) does not make much sense outside of this context (pivot grouping).